eth/tracers/api_blocktrace.go

```diff
@@ -404,6 +412,35 @@ func (api *API) fillBlockTrace(env *traceEnv, block *types.Block) (*types.BlockT
 		txs[i] = types.NewTransactionData(tx, block.NumberU64(), api.backend.ChainConfig())
 	}
 
+	if _, existed := env.Proofs[rcfg.L2MessageQueueAddress.String()]; !existed {
```

Comment:
Under what circumstances would the key `rcfg.L2MessageQueueAddress.String()` exist in the `env.Proofs` map, thus making existed true?

Reply:
For example when `L2Messenger` appends a message to `L2MessageQueue`.